pkg/web/apps/apps_service.go

```diff
@@ -34,3 +35,17 @@ func (a *appsService) GetApps(c echo.Context) (*[]models.App, error) {
 
 	return apps, nil
 }
+
+// GetAppsId retrieves app by id from the repository
+func (a *appsService) GetAppsID(c echo.Context) (*[]models.App, error) {
+
+	apps, err := a.psqlRepository.GetAppsID(c)
+	if err == nil && apps == nil {
+		return nil, httperrors.NotFound(c, "App not found")
+	}
+	if err != nil && apps == nil {
```

Comment:
Should not here be just "err != nil" ? If it has an error then should return internet server error 500. 
The HTTP code 400 which will be returned here is not respecting the HTTP Code response definition. See https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html. The error 400 means that the data sent to the server is not correct and the error here can mean be caused for many things. It is a 500 Internal Server Error.